llvm/lib/Transforms/Utils/SSAUpdaterBulk.cpp

```diff
@@ -143,34 +136,65 @@ void SSAUpdaterBulk::RewriteAllUses(DominatorTree *DT,
     SmallVector<BasicBlock *, 32> IDFBlocks;
     SmallPtrSet<BasicBlock *, 32> LiveInBlocks;
     ComputeLiveInBlocks(UsingBlocks, DefBlocks, LiveInBlocks, PredCache);
-    IDF.resetLiveInBlocks();
     IDF.setLiveInBlocks(LiveInBlocks);
     IDF.calculate(IDFBlocks);
 
+    // Important: reserve sufficient buckets to prevent map growth. [1]
+    BBInfos.init(LiveInBlocks.size() + DefBlocks.size());
+
+    for (auto [BB, V] : R.Defines)
```

Comment:
```suggestion
    for (const auto &[BB, V] : R.Defines)
```

Reply:
BB and V are pointers, I don't need a const ref to a pointer